src/cluster.c

```diff
@@ -5493,6 +5551,10 @@ NULL
                c->argc == 3)
     {
         /* CLUSTER COUNT-FAILURE-REPORTS <NODE ID> */
+        if (verifyClusterNodeName(c->argv[2]->ptr, sdslen(c->argv[2]->ptr)) != C_OK) {
+            addReplyErrorFormat(c, "Invalid cluster node %s", (char *)c->argv[2]->ptr);
+            return;
+        }
```

Comment:
There are a lot more cases here than I was expecting. I think instead we should add the validation logic in `clusterLookupNode`, and have it just return NULL if verifyClusterNodeName fails. Realistically it's not critical that we throw a specific error to end users that they gave an invalid ID, since it's also true that there was node with the given ID.

Reply:
@madolson Indeed, there are a lot of code changes. In some scenarios, if the `clusterLookupNode` returns `NULL`, redis will create a new node(eg. `clusterLoadConfig`), so, I choose to check the format of node name independently

Reply:
I agree that we still need to do extra checks in clusterLoadConfigs, since it has slightly different behavior.